libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -11,25 +11,28 @@
 #ifndef __CCCL_COMPILER_H
 #define __CCCL_COMPILER_H
 
-// Determine the host compiler
-#if defined(__INTEL_LLVM_COMPILER)
-#  define _CCCL_COMPILER_ICC_LLVM
-#elif defined(__INTEL_COMPILER)
+// Determine the host compiler and its version
+#if defined(__INTEL_COMPILER)
 #  define _CCCL_COMPILER_ICC
 #elif defined(__NVCOMPILER)
 #  define _CCCL_COMPILER_NVHPC
+#  define _CCCL_COMPILER_NVHPC_VERSION \
+    (__NVCOMPILER_MAJOR__ * 10000 + __NVCOMPILER_MINOR__ * 100 + __NVCOMPILER_PATCHLEVEL__)
+// __NVCOMPILER_MAJOR__ is 99 in dev builds
+static_assert(__NVCOMPILER_MAJOR__ == 99 || __NVCOMPILER_PATCHLEVEL__ < 100,
+              "__NVCOMPILER_PATCHLEVEL__ must be less than 100");
 #elif defined(__clang__)
 #  define _CCCL_COMPILER_CLANG
 #  define _CCCL_CLANG_VERSION (__clang_major__ * 10000 + __clang_minor__ * 100 + __clang_patchlevel__)
+static_assert(__clang_patchlevel__ < 100, "__clang_patchlevel__ must be less than 100");
```

Comment:
I am not a fan of these asserts, they are outside of our control and will give problems, as seen with the nhc++ issue.

Reply:
I added the patch level checks after seeing strange values for the NVHPC compiler. However, after David's comment, it makes sense to remove them.